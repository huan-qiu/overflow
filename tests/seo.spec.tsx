import React from 'react';
import { render } from 'enzyme';
import Overflow from '../src';
import type { OverflowProps } from '../src';

interface ItemType {
  label: React.ReactNode;
  key: React.Key;
}

interface CaseConf {
  name: string;
  dataLength: number;
  maxCount: OverflowProps<ItemType>['maxCount'];
  suffix?: boolean;
}

function renderItem(item: ItemType) {
  return item.label;
}

describe('Overflow.SEO', () => {
  function getData(count: number) {
    return new Array(count).fill(undefined).map((_, index) => ({
      label: `Label ${index}`,
      key: `k-${index}`,
    }));
  }

  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  const testCases: CaseConf[] = [
    {
      name: 'responsive',
      dataLength: 2,
      maxCount: 'responsive',
    },
    {
      name: 'responsive with suffix',
      dataLength: 2,
      maxCount: 'responsive',
      suffix: true,
    },
    {
      name: 'maxCount number with suffix',
      dataLength: 6,
      maxCount: 4,
      suffix: true,
    },
    {
      name: 'invalidate number with suffix',
      dataLength: 4,
      maxCount: 'invalidate',
      suffix: true,
    },
  ];

  testCases.forEach(({ name, dataLength, maxCount: maxCountVal, suffix }) => {
    it(`${name}`, () => {
      const wrapper = render(
        <Overflow<ItemType>
          data={getData(dataLength)}
          renderItem={renderItem}
          maxCount={maxCountVal}
          suffix={suffix && <span> I am a suffix </span>}
        />,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
