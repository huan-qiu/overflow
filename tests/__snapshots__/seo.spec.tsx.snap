// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overflow.SEO invalidate number with suffix 1`] = `
<div
  class=""
>
  <div
    class=""
  >
    Label 0
  </div>
  <div
    class=""
  >
    Label 1
  </div>
  <div
    class=""
  >
    Label 2
  </div>
  <div
    class=""
  >
    Label 3
  </div>
  <div
    class="rc-overflow-item-suffix"
  >
    <span>
       I am a suffix 
    </span>
  </div>
</div>
`;

exports[`Overflow.SEO maxCount number with suffix 1`] = `
<div
  class="rc-overflow"
>
  <div
    class="rc-overflow-item"
    style="opacity:1"
  >
    Label 0
  </div>
  <div
    class="rc-overflow-item"
    style="opacity:1"
  >
    Label 1
  </div>
  <div
    class="rc-overflow-item"
    style="opacity:1"
  >
    Label 2
  </div>
  <div
    class="rc-overflow-item"
    style="opacity:1"
  >
    Label 3
  </div>
  <div
    class="rc-overflow-item rc-overflow-item-rest"
    style="opacity:1"
  >
    + 2 ...
  </div>
  <div
    class="rc-overflow-item rc-overflow-item-suffix"
    style="opacity:1"
  >
    <span>
       I am a suffix 
    </span>
  </div>
</div>
`;

exports[`Overflow.SEO responsive 1`] = `
<div
  class="rc-overflow"
>
  <div
    class="rc-overflow-item"
    style="opacity:1;order:0;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    Label 0
  </div>
  <div
    class="rc-overflow-item"
    style="opacity:1;order:1;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    Label 1
  </div>
  <div
    aria-hidden="true"
    class="rc-overflow-item rc-overflow-item-rest"
    style="opacity:0;height:0;overflow-y:hidden;order:9007199254740991;pointer-events:none;position:absolute;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    + 0 ...
  </div>
</div>
`;

exports[`Overflow.SEO responsive with suffix 1`] = `
<div
  class="rc-overflow"
>
  <div
    class="rc-overflow-item"
    style="opacity:1;order:0;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    Label 0
  </div>
  <div
    class="rc-overflow-item"
    style="opacity:1;order:1;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    Label 1
  </div>
  <div
    aria-hidden="true"
    class="rc-overflow-item rc-overflow-item-rest"
    style="opacity:0;height:0;overflow-y:hidden;order:9007199254740991;pointer-events:none;position:absolute;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    + 0 ...
  </div>
  <div
    class="rc-overflow-item rc-overflow-item-suffix"
    style="opacity:1;order:9007199254740991;max-width:0;padding:0;margin:0;border-width:0;overflow-x:hidden"
  >
    <span>
       I am a suffix 
    </span>
  </div>
</div>
`;
